halo/withdraw/wrapper.go

```diff
@@ -45,15 +48,24 @@ func (w *BankWrapper) SendCoinsFromModuleToAccountNoWithdrawal(ctx context.Conte
 
 // UndelegateCoinsFromModuleToAccount intercepts all principal undelegations and
 // creates EVM withdrawal to the user account.
-// TODO(corver): This is unexpected in magellan upgrade, must be implemented in drake.
-func (w *BankWrapper) UndelegateCoinsFromModuleToAccount(ctx context.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error {
-	log.Error(ctx, "Unexpected call to DelegateCoinsFromAccountToModule [BUG]", nil, "sender", senderModule, "recipient", recipientAddr, "amt", amt)
+func (w *BankWrapper) UndelegateCoinsFromModuleToAccount(ctx context.Context, senderModule string, recipientAddr sdk.AccAddress, coins sdk.Coins) error {
+	log.Debug(ctx, "Undelegating coins from module to account", "sender", senderModule, "recipient", recipientAddr, "coins", coins)
 
-	if err := w.Keeper.UndelegateCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, amt); err != nil {
-		return errors.Wrap(err, "undelegate coins from module to account")
+	acc := w.ak.GetModuleAccount(ctx, senderModule)
+	if acc == nil {
+		return errors.New("module account does not exist", "module_name", senderModule)
 	}
 
-	return nil
+	_, ok := acc.(banktypes.VestingAccount)
+	if ok {
+		return errors.New("vesting accounts are not supported")
```

Comment:
same here, also a BUG for us.

Reply:
this check must (also) be done in `SendCoinsFromModuleToAccount`

Reply:
oh, and the account to check for "Vesting or not" isn't the module (since modules cannot be vesting accounts), but rather the `recipient` account